packages/core/src/services/documentLogs/computeDocumentLogs.ts

```diff
@@ -0,0 +1,98 @@
+import { and, desc, eq, isNull, sql } from 'drizzle-orm'
+
+import { Commit, DEFAULT_PAGINATION_SIZE, ErrorableEntity } from '../../browser'
+import { database } from '../../client'
+import { calculateOffset } from '../../lib/pagination/calculateOffset'
+import { DocumentLogsRepository } from '../../repositories'
+import {
+  commits,
+  documentLogs,
+  projects,
+  runErrors,
+  workspaces,
+} from '../../schema'
+import { getCommonQueryConditions } from './computeDocumentLogsWithMetadata'
+
+export function computeDocumentLogsQuery(
+  {
+    workspaceId,
+    documentUuid,
+    draft,
+    allowAnyDraft,
+    page = '1',
+    pageSize = String(DEFAULT_PAGINATION_SIZE),
+  }: {
+    workspaceId: number
+    documentUuid?: string
+    draft?: Commit
+    allowAnyDraft?: boolean
+    page?: string
+    pageSize?: string
+  },
+  db = database,
+) {
+  const repo = new DocumentLogsRepository(workspaceId, db)
+  const offset = calculateOffset(page, pageSize)
+  const scope = repo.scope.as('document_logs')
```

Comment:
This repo is already filtering logs that has 0 `run_errors` associated. In order to make it work I had to do this:
```javascript
const scope = repo.scope.as('document_logs')
  const fields = scope._.selectedFields
  return db
    .select(fields)
    .where(...MY_THINGS...)
```
But in other places I see we're doing this

```javascript
return repo.scope.where(...MY_THINGS...)
```
I'm a bit confused why this second why is not working for me. This issue looks similar to [what I see](https://github.com/drizzle-team/drizzle-orm/issues/1644)